pkg/controller/jobframework/reconciler.go

```diff
@@ -1051,15 +1051,21 @@ func (r *JobReconciler) prepareWorkload(ctx context.Context, job GenericJob, wl
 }
 
 func (r *JobReconciler) extractPriority(ctx context.Context, podSets []kueue.PodSet, job GenericJob) (string, string, int32, error) {
-	if workloadPriorityClass := WorkloadPriorityClassName(job.Object()); len(workloadPriorityClass) > 0 {
-		return utilpriority.GetPriorityFromWorkloadPriorityClass(ctx, r.client, workloadPriorityClass)
-	}
 	if jobWithPriorityClass, isImplemented := job.(JobWithPriorityClass); isImplemented {
-		return utilpriority.GetPriorityFromPriorityClass(
-			ctx, r.client, jobWithPriorityClass.PriorityClass())
+		return ExtractPriority(ctx, r.client, job.Object(), podSets, jobWithPriorityClass.PriorityClass)
+	} else {
+		return ExtractPriority(ctx, r.client, job.Object(), podSets, nil)
+	}
```

Comment:
let's don't duplicate calling `ExtractPriority`. We can extract the function like:

```golang
var customPrioritFunc func()...
if jobWithPriorityClass, isImplemented := job.(JobWithPriorityClass); isImplemented {
   customPrioritFunc = jobWithPriorityClass.PriorityClass
}
return ExtractPriority(..., customPrioritFunc)
```

Reply:
Done